content/en/database_monitoring/setup_mongodb/selfhosted.md

```diff
@@ -0,0 +1,198 @@
+```
+
+Grant addtionnal permissions to the `datadog` user in the databases you want to monitor:
+
+```shell
+db.grantRolesToUser("datadog", [
+  { role: "read", db: "mydatabase" },
+  { role: "read", db: "myotherdatabase" }
+])
+```
+
+Alternatively, you can grant `readAnyDatabase` role to the `datadog` user in the `admin` database to monitor all databases:
+
+```shell
+db.grantRolesToUser("datadog", [
+  { role: "readAnyDatabase", db: "admin" }
+])
+```
+
+{{% /tab %}}
+{{% tab "Replica Set" %}}
+
+In a Mongo shell, authenticate to the primary node of the replica set, create a read-only user for the Datadog Agent in the `admin` database and grant the required permissions:
+
+```shell
+# Authenticate as the admin user.
+use admin
+db.auth("admin", "<YOUR_MONGODB_ADMIN_PASSWORD>")
+
+# Create the user for the Datadog Agent.
+db.createUser({
+  "user": "datadog",
+  "pwd": "<UNIQUEPASSWORD>",
+  "roles": [
+    { role: "read", db: "admin" },
+    { role: "read", db: "local" },
+    { role: "clusterMonitor", db: "admin" }
+  ]
+})
+```
+
+Grant addtionnal permissions to the `datadog` user in the databases you want to monitor:
+
+```shell
+db.grantRolesToUser("datadog", [
+  { role: "read", db: "mydatabase" },
+  { role: "read", db: "myotherdatabase" }
+])
+```
+
+Alternatively, you can grant `readAnyDatabase` role to the `datadog` user in the `admin` database to monitor all databases:
+
+```shell
+db.grantRolesToUser("datadog", [
+  { role: "readAnyDatabase", db: "admin" }
+])
+```
+
+{{% /tab %}}
+{{% tab "Sharded Cluster" %}}
+
+1. For each shard in your cluster, connect to the primary node of the shard, create a read-only user for the Datadog Agent in the `admin` database and grant the required permissions:
+
+```shell
+# Authenticate as the admin user.
+use admin
+db.auth("admin", "<YOUR_MONGODB_ADMIN_PASSWORD>")
+
+# Create the user for the Datadog Agent.
+db.createUser({
+  "user": "datadog",
+  "pwd": "<UNIQUEPASSWORD>",
+  "roles": [
+    { role: "read", db: "admin" },
+    { role: "read", db: "local" },
+    { role: "clusterMonitor", db: "admin" }
+  ]
+})
+```
+
+Grant addtionnal permissions to the `datadog` user in the databases you want to monitor:
+
+```shell
+db.grantRolesToUser("datadog", [
+  { role: "read", db: "mydatabase" },
+  { role: "read", db: "myotherdatabase" }
+])
+```
+
+Alternatively, you can grant `readAnyDatabase` role to the `datadog` user in the `admin` database to monitor all databases:
+
+```shell
+db.grantRolesToUser("datadog", [
+  { role: "readAnyDatabase", db: "admin" }
+])
+```
+
+2. Follow the same steps and create the same user from a mongos proxy. This action creates the local user in the config servers and allows direct connection.
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Install and configure the Agent
+
+It's recommended to install the agent directly on the MongoDB host as that enables the agent to collect a variety of system telemetry (CPU, memory, disk, network) in addition to MongoDB specific telemetry.
```

Comment:
It's recommended to install the `A`gent directly on the MongoDB host as that enables the `A`gent to collect a variety of system